src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.cs

```diff
@@ -320,7 +326,7 @@ internal async Task ProcessMessageAsync(CloudQueueMessage message, TimeSpan visi
                     await timer.StopAsync(cancellationToken);
                 }
 
-                await _queueProcessor.CompleteProcessingMessageAsync(message, result, cancellationToken);
+                await _queueProcessor.CompleteProcessingMessageAsync(message, result, _shutdownCancellationTokenSource.Token);
```

Comment:
One nit -- can you add a comment above this explaining why we're using a different token here? I know we're all going to forget why we did this in about 2 weeks :-)

Reply:
Good idea - sure.